core/src/test/java/org/jdbi/v3/TestForeman.java

```diff
@@ -13,51 +13,79 @@
  */
 package org.jdbi.v3;
 
-
+import static org.easymock.EasyMock.replay;
 import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertSame;
 
 import java.sql.PreparedStatement;
 import java.sql.SQLException;
+import java.sql.Types;
 
+import org.easymock.EasyMock;
+import org.easymock.EasyMockSupport;
 import org.jdbi.v3.tweak.Argument;
 import org.jdbi.v3.tweak.ArgumentFactory;
+import org.junit.After;
 import org.junit.Test;
 
-public class TestForeman
+public class TestForeman extends EasyMockSupport
 {
-    @Test
-    public void testWaffling()
-    {
-        final Foreman foreman = new Foreman();
+    private static final String I_AM_A_STRING = "I am a String";
 
-        final Argument longArgument = foreman.waffle(Object.class, new Long(3L), null);
-        assertSame(LongArgument.class, longArgument.getClass());
+    private final Foreman foreman = new Foreman();
 
-        final Argument shortArgument = foreman.waffle(Object.class, (short) 2000, null);
-        assertSame(ShortArgument.class, shortArgument.getClass());
+    public PreparedStatement stmt = createMock(PreparedStatement.class);
 
-        final Argument stringArgument = foreman.waffle(Object.class, "I am a String!", null);
-        assertSame(StringArgument.class, stringArgument.getClass());
+    @After
+    public void verify() {
+        verifyAll();
     }
 
     @Test
-    public void testExplicitWaffling()
+    public void testWaffleLong() throws Exception
     {
-        final Foreman foreman = new Foreman();
+        stmt.setLong(1, 3);
+        replay(stmt);
+        foreman.waffle(Object.class, new Long(3L), null).apply(1, stmt, null);
```

Comment:
Nice, that definitely is better.  I guess I'm convinced enough to try Mockito out, the EasyMockRule isn't very impressive.